morello/benches/update_for_tiling.rs

```diff
@@ -1,13 +1,20 @@
 use iai_callgrind::{black_box, main};
+use nonzero::nonzero as nz;
 use smallvec::smallvec;
 
 use morello::layout::Layout;
+use morello::shape;
 
 #[inline(never)]
 fn update_for_tiling() {
-    let shape = [64, 64, 64];
-    let tile_shape = [64, 8, 8];
-    let layout = Layout::New(smallvec![(0, None), (1, None), (2, None), (1, Some(8))]);
+    let shape = shape![64, 64, 64];
+    let tile_shape = shape![64, 8, 8];
+    let layout = Layout::New(smallvec![
```

Comment:
@samkaufman  Could you please explain how `Layout::New` and `Layout::new` are different (when to use)?  I understand `Layout::new` has many assertions, but I did not get how to determine which to use.  I implemented the `layout!` macro, which uses `Layout::new`, but I was curious if we could replace `Layout::New` with the macro.

Reply:
@samkaufman  I see.  I will replace `Layout::New` calls with `layout!`.  Also, do you mind if I create a new PR that removes `Layout::New` and makes `Layout` just a struct?